scripts/create_multi_session_df.py

```diff
@@ -19,90 +19,89 @@
+
+    behavior_conditions = [['ophys_experiment_id', 'omitted'],
+                            ['ophys_experiment_id', 'omitted', 'epoch'],
+                            ['ophys_experiment_id', 'is_change'],
+                            ['ophys_experiment_id', 'is_change', 'epoch'],
+                            ['ophys_experiment_id', 'is_change', 'image_name'],
+                            ['ophys_experiment_id', 'is_change', 'image_name', 'epoch'],
+                            ['ophys_experiment_id', 'is_change', 'hit'],
+                            ['ophys_experiment_id', 'is_change', 'pre_change', 'epoch'],
+                            ['ophys_experiment_id', 'is_change', 'hit', 'epoch'],
+                            ['cell_specimen_id', 'omitted', 'pre_omitted'],]
+
+
+    # event types corresponding to the above physio and behavior conditions - must be in same sequential order
+    event_types_for_conditions = ['omissions', 'omissions',
+                                  'changes', 'changes',
+                                  'changes', 'changes', 'changes',
+                                  'all', 'all', 'all']
+
+    # add engagement state to all conditions
+    # for i in range(len(physio_conditions)):
+    #     physio_conditions[i].insert(1, 'engagement_state')
+    #     behavior_conditions[i].insert(1, 'engagement_state')
+
+    # create dfs for all data types and conditions for physio data
+    for data_type in physio_data_types:
+    # data_type = 'events'
+        for i, conditions in enumerate(physio_conditions):
+            print(conditions)
+            event_type = event_types_for_conditions[i]
+            if event_type == 'omissions':
+                response_window_duration = 0.75
+            else:
+                response_window_duration = 0.5
+            print('creating multi_session_df for', data_type, event_type, conditions)
+            try: # use try except so that it skips over any conditions that fail to generate for some reason
+                df = io.get_multi_session_df(project_code, session_number, conditions, data_type, event_type,
+                                             time_window=time_window, interpolate=interpolate, output_sampling_rate=output_sampling_rate,
+                                             response_window_duration=response_window_duration,
+                                             use_extended_stimulus_presentations=use_extended_stimulus_presentations,
+                                             overwrite=False)
+            except Exception as e:
+                print('failed to create multi_session_df for', data_type, event_type, conditions)
+                print(e)
+
+
+    # create dfs for all data types and conditions for behavior data
+    for data_type in behavior_data_types:
+        for i, conditions in enumerate(behavior_conditions):
+            event_type = event_types_for_conditions[i]
+            if event_type == 'omissions':
```

Comment:
why is omission window different from response window?

Reply:
because i wanted to average over a longer window?

Reply:
@yavorska-iryna for omissions, i want to average over the full 750ms window because most of the response is right before the next image because of the ramp. for regular stimulus presentations, i just want the stimulus response, so 500ms captures the stimulus window plus 250ms more in the case where cells have off responses